docs/rules-engine/how-to-use/README.md

```diff
@@ -0,0 +1,218 @@
+
+If the Otter rules engine comes with built-in actions, operators and one integrated fact, you will need your own custom
+facts and maybe add your own custom operators and actions.
+
+Check out the dedicated sections:
+* [Create custom facts](./custom-fact.md)
+* [Create custom operators](./custom-operator.md)
+* [Create custom action](./custom-action.md)
+
+### How to measure your Ruleset performances
+
+If you want to measure the performance of your Rulesets, the Otter rules engine exposes a set of metrics.
+Learn more about them in the [performance documentation](./performance-metrics.md).
+
+### Examples of classic Rulesets
+
+Check out some classic examples to build your first Ruleset:
+* [Basic Ruleset](../examples/basic-rule.md)
+* [Input fact as a complex object](../examples/complex-fact.md)
+* [Rules with nested conditions](../examples/nested-conditions.md)
+
+### Complex Rulesets 
+
+#### Runtime facts
+
+An action can result in the update of a runtime variable scoped within the Ruleset. Other rules of the same Ruleset can
+then make reference to the runtime variable.
+
+This can lead to factorization as it is then possible to use runtime facts as a way to represent a complex condition tree.
+You can find an example of this feature in the [runtime fact documentation](../examples/runtime-facts.md).
+
+#### Implicit built-in facts - the case of dateInNextMinutes and dateNotInNextMinutes
+
+While it is the norm and recommendation to keep the operators of rules [pure](https://en.wikipedia.org/wiki/Pure_function),
+there might be cases where a pure operator does not make any sense.
+
+This is the case of `dateInNextMinutes` and `dateNotInNextMinutes`.
+
+In those use cases, the reference to a current time fact is implicit and a current time fact cannot really be used as an
+explicit parameter.
+For this reason, the Otter framework introduces the concept of implicit 'built-in' facts.
+
+Check out how to integrate those operators and override the default implicit fact behavior in the 
+[built-in fact documentation](./built-in-facts.md).
+
+#### How to enable ruleset only when the affected components is present
+
+If your Ruleset only affects a component to modify its configuration or one of its translation key, you won't need it to
+run on pages where this component is missing. You can restrict the evaluation of your Ruleset to only pages with your
+component thanks to the `linkedComponents` property and improve the performances of your application.
+
+Check out the [linkedComponents documentation](./dedicated-component-ruleset.md).
+
+#### How to use query facts
+
+Let's consider an application that can be directly targeted via a deeplink. The parameters of the deeplink will be used 
+as the inputs of your rules, and they will help deliver a different user experience based on the deeplink.
+This is something you may need for A/B testing or to apply a customization based on the website used as a portal to your
```

Comment:
For example, you can leverage this to do A/B testing in your application, sending a parameter with 2 different values and creating rules based on this fact's value different parameters.